pkg/controller/ovn_ic.go

```diff
@@ -323,40 +334,39 @@ func (c *Controller) startOvnIC(icHost, icNbPort, icSbPort string) error {
 	}
 	output, err := cmd.CombinedOutput()
 	if err != nil {
-		klog.Error(err)
 		return fmt.Errorf("%s", output)
 	}
 	return nil
 }
 
-func (c *Controller) stopOvnIC() error {
+func (c *Controller) stopOVNIC() error {
 	cmd := exec.Command("/usr/share/ovn/scripts/ovn-ctl", "stop_ic")
 	output, err := cmd.CombinedOutput()
 	if err != nil {
-		klog.Error(err)
 		return fmt.Errorf("%s", output)
 	}
 	return nil
 }
 
-func (c *Controller) waitTsReady() error {
-	retry := 6
-	for retry > 0 {
-		ready, err := c.allSubnetReady(util.InterconnectionSwitch)
-		if err != nil {
-			klog.Error(err)
-			return err
-		}
-
-		if ready {
-			return nil
-		}
-
-		klog.Info("wait for logical switch %s ready", util.InterconnectionSwitch)
-		time.Sleep(5 * time.Second)
-		retry--
+func (c *Controller) createTS(icNBAddr, tsName, subnet string) error {
+	cmd := exec.Command("ovn-ic-nbctl",
+		fmt.Sprintf("--db=%s", icNBAddr),
+		ovs.MayExist, "ts-add", tsName,
+		"--", "set", "Transit_Switch", tsName, fmt.Sprintf(`external_ids:subnet="%s"`, subnet))
+	if os.Getenv("ENABLE_SSL") == "true" {
+		cmd = exec.Command("ovn-ic-nbctl",
+			fmt.Sprintf("--db=%s", icNBAddr),
+			"--private-key=/var/run/tls/key",
+			"--certificate=/var/run/tls/cert",
+			"--ca-cert=/var/run/tls/cacert",
+			ovs.MayExist, "ts-add", tsName,
+			"--", "set", "Transit_Switch", tsName, fmt.Sprintf(`external_ids:subnet="%s"`, subnet))
+	}
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%s", output)
 	}
```

Comment:
`err` should be logged.

Reply:
err is displayed in the function that calls it

Reply:
1